src/PaymentGateways/PayPalCommerce/PayPalPayment.php

```diff
@@ -0,0 +1,57 @@
+<?php
+namespace Give\PaymentGateways\PayPalCommerce;
+
+use Give\Helpers\ArrayDataSet;
+use InvalidArgumentException;
+
+/**
+ * Class PayPalPayment
+ * @package Give\PaymentGateways\PayPalCommerce
+ *
+ * @property-read string $id
+ * @property-read string $status
+ * @property-read string $amount
+ * @property-read string $createTime
+ * @property-read string $updateTime
+ */
+class PayPalPayment {
+	/**
+	 * Create PayPalPayment object from given array.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param $array
+	 *
+	 * @return PayPalPayment
+	 */
+	public static function fromArray( $array ) {
+		$payment = new static();
+
+		$payment->validate( $array );
+
+		$array = ArrayDataSet::camelCaseKeys( $array );
+
+		foreach ( $array as $itemName => $value ) {
+			$payment->{$itemName} = $value;
+		}
+
+		return $payment;
+	}
+
+	/**
+	 * Return whether array has validate format or not.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param array $array
+	 * @throws InvalidArgumentException
+	 */
+	private function validate( $array ) {
+		$required = [ 'id', 'amount', 'status', 'create_time', 'update_time', 'links' ];
```

Comment:
Same thing as the last class; please make sure to add the properties to the class so they're clear.

Reply:
fixed: https://github.com/impress-org/givewp/compare/f549508bd6a0...ae1fb915985b